go/tasks/v1/flytek8s/plugin_executor.go

```diff
@@ -283,10 +312,18 @@ func (e *K8sTaskExecutor) CheckTaskStatus(ctx context.Context, taskCtx types.Tas
 	// This must happen after sending admin event. It's safe against partial failures because if the event failed, we will
 	// simply retry in the next round. If the event succeeded but this failed, we will try again the next round to send
 	// the same event (idempotent) and then come here again...
-	if finalStatus.Phase.IsTerminal() && len(o.GetFinalizers()) > 0 {
-		err = e.ClearFinalizers(ctx, o)
-		if err != nil {
-			return types.TaskStatusUndefined, err
+	if finalStatus.Phase.IsTerminal() {
+		if len(o.GetFinalizers()) > 0 {
+			err = e.ClearFinalizers(ctx, o)
+			if err != nil {
+				return types.TaskStatusUndefined, err
+			}
+		}
+
+		finalStatus = types.TaskStatus{
+			Phase:        taskCtx.GetPhase(),
+			PhaseVersion: taskCtx.GetPhaseVersion(),
+			State:        storeK8sObjectStatus(k8sObjectDeleted, finalStatus.Phase),
```

Comment:
The actual handler might be using a state on its own? I guess it's fine since we are deprecating this code anyway...

Reply:
I checked, none of the handlers today are :). It will go away soon.